sdk/storage/Azure.Storage.Blobs/tests/PageBlobClientTests.cs

```diff
@@ -501,6 +501,38 @@ await blob.CreateAsync(blobSize)
             TestHelper.AssertSequenceEqual(data, actual.ToArray());
         }
 
+        [LiveOnly]
+        [Test]
+        public async Task UploadPagesAsync_ProgressReporting()
+        {
+            await using DisposingContainer test = await GetTestContainerAsync();
+
+            // Arrange
+            long blobSize = 4 * Constants.MB;
+            PageBlobClient blob = await CreatePageBlobClientAsync(test.Container, blobSize);
+            var data = GetRandomBuffer(blobSize);
+            Progress progress = new Progress();
+
+            using (var stream = new MemoryStream(data))
+            {
+                // Act
+                await blob.UploadPagesAsync(
+                    content: stream,
+                    offset: 0,
+                    progressHandler: progress);
+            }
+
+            // Assert
+            Assert.IsFalse(progress.List.Count == 0);
```

Comment:
nit: this assert is not needed since we have the assert on the final element.

Reply:
I think it makes it more obvious what the issue is if the test fails.